### ISSUE TITLE
[Question] ChewableSorter and results array filled with numbers?

### ISSUE THREAD

**Issue by Runisco** [ussjohnjay/sttat#1]
I don't know if you're interested in answering questions like this, but I am really spinning in the mud right now. 

I am currently at the point of trying to write ChewableSorter into python. And I am getting errors which seem to relate back to the arrays in results not containing any numbers.

The code that starts everything is this:
```javascript
var results = [];  
var resultsRefillCostTotal = [];  
for (var iExtend = 0; iExtend <= numExtends; ++iExtend) {  
     results.push([]);
     results[iExtend].length = numSims;
     resultsRefillCostTotal.push(0);}
```

Now in my python code I am able to generate lists inside lists. Either 1000 or 5000 depending on the other code. 
However, if I try to console.log the results from your file, it is filled with small incremental numbers. And I cannot figure out where those numbers come from.

Which means when I get to 
```javascript
# calculate and display results
for extend in range(0,num_extends):
ex_results = results[extend]
```
All that is happening is ex_results becomes a long list of empty lists. Which means that
```javascript
dilemma = math.floor(ex_results[i]/hoursBetweenDilemmas)
```
fails because it's trying to do a math operation with an empty list and an int

Where are those numbers coming from?

**Comment by ussjohnjay** [comment]
First off, I'm far from an expert on this part of the voyage calculator. I just use it with the happy assumption that Chewable knows what it's doing.

That said, `ex_results` should be filled with voyage times calculated from each of the 5000 simulation runs. In the JS script that I think you're using, the simulations start on line 107 (after the loop you excerpted in your comment):

```
for (var iSim = 0; iSim < numSims; iSim++)
``` 

A whole bunch of math happens to calculate the voyage time for each sim and the time then gets assigned to `results` on line 171:

```
results[extend][iSim] = tick/ticksPerHour;
```

Did you adapt this loop into Python already?

And for clarification, the `voyTime` variable inside the sim loop is the time calculated by its respective simulation. The `voyTime` in the later part of the script is actually the median of all the simulations.

**Comment by Runisco** [comment]
In the end several things seem to have gone wrong. 

The results list ended up with several sets without any numbers, only results[0][iSim] were filled with numbers.  
Whether this caused the numbers to be in the range of 20 and up instead of 2 and up is unknown. 

It also took a lot longer than your webpage, which is either on python, or errors in code by me. 
I also did not get back the crew as suggested by the page, nor the suggestions the page considered. 

In the end, I've decided to delete absolutely everything, and try again in a proper python IDE. I might show up with more questions later and hope you'll be both able and willing to provide insight. 

But thank you for responding to me :)
